### PR TITLE
jsonpb: fix Unmarshal when single enum value is provided for a repeated enum field

### PR DESCRIPTION
**Pull request by maditya**
…d enum field

### CODE REVIEW

**Comment by maditya** [comment]
ping @bcmills

File: jsonpb/jsonpb.go [hunk 1]
```diff
@@ -634,6 +634,9 @@ func (u *Unmarshaler) unmarshalValue(target reflect.Value, inputValue json.RawMe
 			target.Set(reflect.New(targetType.Elem()))
 			target = target.Elem()
 		}
+		if target.Kind() != reflect.Int32 {
```
Comment (hyangah) [inline jsonpb/jsonpb.go]: Instead this, in line 624, you can check whether the target can be an enum type.

maybeEnum := target.Kind() == reflect.Int32 ||
  target.Kind() == reflect.Ptr && targetType.Elem().Kind() == reflect.Int32  // proto2
if maybeEnum && inputValue[0] == '"' && prop != nil && prop.Enum != "" {
 ...
}

The advantage is the case this change addresses will be handled by the json unmarshal code so the error message will be a bit clearer.

"json: cannot unmarshal string into Go value of type []json.RawMessage"
vs. 
"mismatched type, want int32, got slice"

**Comment by cybrcodr** [comment]
I'd like to follow up on this PR.  It would be nice to have this merged in if you can follow through the review comment.  Thanks.

**Comment by dsnet** [comment]
Apparently this is a duplicate of #627, which was recently merged.

**Pull request closed without merging**